src/library/coverartcache.cpp

```diff
@@ -170,12 +162,28 @@ void CoverArtCache::coverLoaded() {
         qDebug() << "CoverArtCache::coverLoaded" << res.cover;
     }
 
-    QPixmap pixmap = cacheCover(res.cover, res.desiredWidth);
+    // Don't cache full size covers (resizedToWidth = 0)
+    // Large cover art wastes space in our cache and will likely
+    // uncache a lot of the small covers we need in the library
+    // table.
+    // Full size covers are used in the Skin Widgets, which are
+    // loaded with an artificial delay anyway and an additional
+    // re-load delay can be accepted.
+
+    // Create pixmap, GUI thread only 
+    QPixmap pixmap = QPixmap::fromImage(res.cover.image);
+    if (!pixmap.isNull() && res.cover.resizedToWidth != 0) {
```

Comment:
Good. The null check placed in front of the actual condition like here is easier to read.